postgraphiql/src/components/PostGraphiQL.js

```diff
@@ -209,6 +241,21 @@ class PostGraphiQL extends React.PureComponent {
     }
   };
 
+  updateHeadersState(newHeaderText) {
+     this.setState(
+       {
+         headersText: newHeaderText,
+         headersTextValid: isValidJSON(newHeaderText),
+       },
+       () => {
+         if (this.state.headersTextValid && this.subscriptionsClient) {
+           // Reconnect to websocket with new headers
+           this.subscriptionsClient.close(false, true);
+         }
+       }
+     )
```

Comment:
This'd be a lot neater if we refactored everything for React hooks. (Do not action this comment!)